daprdocs/content/en/developing-applications/ides/visual-studio-code/vscode-dapr-extension.md

```diff
@@ -0,0 +1,136 @@
+---
+type: docs
+title: "Dapr Visual Studio Code extension overview"
+linkTitle: "Overview"
+weight: 10000
+description:  "How to develop and run Dapr applications with the Dapr extension"
+---
+
+
+Dapr offers a *preview* [Dapr Visual Studio Code extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-dapr) for local development and debugging of your Dapr applications.
+
+<a href="vscode:extension/ms-azuretools.vscode-dapr" class="btn btn-primary" role="button">Open in VSCode</a>
+
+## Features
+
+### Scaffold Dapr tasks 
+
+* Helps scaffold VS Code task.json and launch.json configurations needed to debug your application within the Dapr environment.
+* Requires an already existing launch.json files to be found in the VS Code workspace.
+* Example NodeJS app:
+
+
+Before scaffolding version of launch.json configurations:
+```json
+{
+     "version": "0.2.0",
+     "configurations": [
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js"
+        }
+}
+```
+After scaffolding version of launch.json:
+```json
+{
+     "version": "0.2.0",
+     "configurations": [
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js"
+        },
+        {
+            "type": "pwa-node",
+            "request": "launch",
+            "name": "Launch Program with Dapr",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/app.js",
+            "preLaunchTask": "daprd-debug",
+            "postDebugTask": "daprd-down"
+        }
+}
+```
+After scaffolding task.json:
+
+```json
+{
+	"version": "2.0.0",
+	"tasks": [
+		{
+			"appId": "nodeapp",
+			"appPort": 3500,
+			"label": "daprd-debug",
+			"type": "daprd"
+		},
+		{
+			"appId": "nodeapp",
+			"label": "daprd-down",
+			"type": "daprd-down"
+		}
+	]
+}
+```
```

Comment:
Way too verbose. I suggest dropping this.

Reply:
See my open comments for my suggestions

Reply:
I can just leave, "daprd-debug" for the task.json and  the "Launch Program with Dapr" content then. that way each JSON example is only like 4 to 6 lines.